### PR TITLE
Add tags to virtual network creation

### PR DESCRIPTION
**Pull request by nacx**

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ protected synchronized void createDefaultNetworkIfNeeded(String group, String lo
          logger.debug(">> network options have not been configured. Creating network %s(%s) and subnet %s(%s)", name,
                defaultVnetAddressPrefix, name, defaultSubnetAddressPrefix);
          
-         api.getVirtualNetworkApi(options.getResourceGroup()).createOrUpdate(name, location, properties);
+         api.getVirtualNetworkApi(options.getResourceGroup()).createOrUpdate(name, location, null, properties);
```
Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java]: out of curiosity, how do you plan to use the `tags` is the `CreateResourceThenCreateNodes` has them set to null?

Comment (nacx) [reply azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java]: Using the provider-specific API. Without this change, there is no way to configure the tags for a network.

Comment (andreaturli) [reply azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java]: sure, that makes sense!
would it make sense to think about a default strategy to tag the `virtualNetwork` so the clean up resources step can take advantage of it?

Comment (nacx) [reply azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java]: Yes, We could do that and follow the same approach than we use for the availability sets (although leaving a default network that will be reused per resource group is not a big deal). Anyway, stuff for a different PR, right?

Comment (andreaturli) [reply azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java]: yes, for a subsequent PR!

**Comment by nacx** [comment]
Pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/c9cd6eaf) and [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/fd22d3ad).

**Pull request closed without merging**